sql/core/src/test/java/test/org/apache/spark/sql/JavaDataFrameSuite.java

```diff
@@ -303,27 +305,42 @@ public void pivot() {
     Assert.assertEquals(30000.0, actual.get(1).getDouble(2), 0.01);
   }
 
+  private String getResource(String resource) {
+    try {
+      // The following "getResource" has different behaviors in SBT and Maven.
+      // When running in Jenkins, the file path may contain "@" when there are multiple
+      // SparkPullRequestBuilders running in the same worker
+      // (e.g., /home/jenkins/workspace/SparkPullRequestBuilder@2)
+      // When running in SBT, "@" in the file path will be returned as "@", however,
+      // when running in Maven, "@" will be encoded as "%40".
```

Comment:
Here are the debug output I tried in Jenkins:

```
Running test.org.apache.spark.sql.JavaDataFrameSuite
raw:file:/home/jenkins/workspace/SparkPullRequestBuilder%402/sql/core/target/scala-2.11/test-classes/text-suite.txt
Find: /home/jenkins/workspace/SparkPullRequestBuilder@2/sql/core/target/scala-2.11/test-classes/text-suite.txt
raw:file:/home/jenkins/workspace/SparkPullRequestBuilder%402/sql/core/target/scala-2.11/test-classes/text-suite.txt
Find: /home/jenkins/workspace/SparkPullRequestBuilder@2/sql/core/target/scala-2.11/test-classes/text-suite.txt
raw:file:/home/jenkins/workspace/SparkPullRequestBuilder%402/sql/core/target/scala-2.11/test-classes/text-suite.txt
Find: /home/jenkins/workspace/SparkPullRequestBuilder@2/sql/core/target/scala-2.11/test-classes/text-suite.txt
raw:file:/home/jenkins/workspace/SparkPullRequestBuilder%402/sql/core/target/scala-2.11/test-classes/text-suite.txt
Find: /home/jenkins/workspace/SparkPullRequestBuilder@2/sql/core/target/scala-2.11/test-classes/text-suite.txt
Tests run: 17, Failures: 0, Errors: 2, Skipped: 0, Time elapsed: 7.628 sec <<< FAILURE! - in test.org.apache.spark.sql.JavaDataFrameSuite
testTextLoad(test.org.apache.spark.sql.JavaDataFrameSuite)  Time elapsed: 0.356 sec  <<< ERROR!
org.apache.spark.sql.AnalysisException: Path does not exist: file:/home/jenkins/workspace/SparkPullRequestBuilder%402/sql/core/target/scala-2.11/test-classes/text-suite.txt;
	at test.org.apache.spark.sql.JavaDataFrameSuite.testTextLoad(JavaDataFrameSuite.java:349)

testGenericLoad(test.org.apache.spark.sql.JavaDataFrameSuite)  Time elapsed: 0.322 sec  <<< ERROR!
org.apache.spark.sql.AnalysisException: Path does not exist: file:/home/jenkins/workspace/SparkPullRequestBuilder%402/sql/core/target/scala-2.11/test-classes/text-suite.txt;
	at test.org.apache.spark.sql.JavaDataFrameSuite.testGenericLoad(JavaDataFrameSuite.java:311)

Running test.org.apache.spark.sql.JavaRowSuite
Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 sec - in test.org.apache.spark.sql.JavaRowSuite

Results :

Tests in error: 
  JavaDataFrameSuite.testGenericLoad:311 ? Analysis Path does not exist: file:/h...
  JavaDataFrameSuite.testTextLoad:349 ? Analysis Path does not exist: file:/home...
```
This is the line returned by `Thread.currentThread().getContextClassLoader().getResource("text-suite.txt").toString()`
``` 
file:/home/jenkins/workspace/SparkPullRequestBuilder%402/sql/core/target/scala-2.11/test-classes/text-suite.txt
```